app/models/king.rb

```diff
@@ -6,4 +6,80 @@ def valid_move?(position_2)
 			return false
 		end
 	end
+
+	def checked?(position)
```

Comment:
Rather than the same code copied and pasted I would implement this like this:

```
opposite_color = {'black' => 'white', 'white' => 'black'}
self.game.pieces.where(:color => opposite_color[self.color]).find_each do |piece|
if piece.valid_move?(position)
	return true
end
end
```